service/cziti/sdk.go

```diff
@@ -560,36 +561,41 @@ func eventCB(ztx C.ziti_context, event *C.ziti_event_t) {
 		var maximumTimeout int32 = -1
 		noTimeoutSvc := false
 		if len(servicesToAdd) > 0 {
-			for _, svc := range servicesToAdd {
-				if svc.Timeout >= 0 {
-					if minimumTimeout == -1 || minimumTimeout > svc.Timeout {
-						minimumTimeout = svc.Timeout
+			zid.Services.Range(func(key interface{}, value interface{}) bool {
```

Comment:
do you think it makes the most sense to do the min/max determination while iterating through the changed and added loops above (lines 513 to 559)? That way you don't need to iterate all the services two times - just the one loop that's already being done. I thought of this because I noticed you changed from using 'servicesToAdd' to 'zid.Services'. That made me wonder if zid.Services was updated at this point which I can't remember if it was or not.

Reply:
While testing, I noticed that the service events for all the services are not getting triggered, when mfa timeout changes. In that case, we will have the entire services only in the zid object. So I looped thought the zid services loop after the updated/added events.